clustering/marathon.py

```diff
@@ -0,0 +1,555 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = """
+module: marathon_app
+version_added: "2.1"
+short_description: start and stop applications with Marathon
+description:
+  - Start and stop applications with Marathon.
+author: "Ludovic Claude (@ludovicc)"
+
+options:
+  uri:
+    required: true
+    description:
+      - Base URI for the Marathon instance
+
+  state:
+    required: true
+    choices: [ present, absent, restart, kill ]
+    default: "present"
+    description:
+      - The operation to perform.
+
+  username:
+    required: false
+    description:
+      - The username to log-in with.
+
+  password:
+    required: false
+    description:
+      - The password to log-in with.
+
+  id:
+    required: true
+    description:
+      - Unique identifier for the app consisting of a series of names separated by slashes.
+
+  command:
+    aliases: [ cmd ]
+    required: false
+    description:
+      - The command that is executed.
+
+  arguments:
+    aliases: [ args ]
+    required: false
+    description:
+      - An array of strings that represents an alternative mode of specifying the command to run.
+
+  cpus:
+    required: false
+    description:
+     - The number of CPU`s this application needs per instance. This number does not have to be integer, but can be a fraction.
+
+  memory:
+    aliases: [ mem ]
+    required: false
+    description:
+     - The amount of memory in MB that is needed for the application per instance.
+
+  disk:
+    required: false
+    description:
+     - The amount of disk in MB that is reserved for the application per instance.
+
+  ports:
+    required: false
+    description:
+     - An array of required port resources on the host.
+
+  requirePorts:
+    required: false
+    description:
+     - If true, the ports you have specified are used as host ports.
+
+  instances:
+    required: false
+    description:
+     - The number of instances of this application to start.
+
+  executor:
+    required: false
+    description:
+     - The executor to use to launch this application.
+
+  user:
+    required: false
+    description:
+      - The user to use to launch this application.
+
+  container:
+    required: false
+    description:
+     - Additional data passed to the containerizer on application launch. This is a free-form data structure that can contain arbitrary data.
+
+  docker_image:
+    required: false
```

Comment:
if required: false is defined you must define a default e.g. default: null also see http://docs.ansible.com/ansible/developing_modules.html#module-checklist